Gems/AudioSystem/Code/Source/Editor/AudioControl.h

```diff
@@ -11,29 +11,64 @@
 
 #include <ACETypes.h>
 #include <AzCore/std/string/string_view.h>
+#include <AzCore/XML/rapidxml.h>
 
 #include <IAudioConnection.h>
 #include <IAudioSystemControl.h>
 
-#include <ISystem.h>
-#include <IXml.h>
-
 namespace AudioControls
 {
     class CATLControlsModel;
 
     //-------------------------------------------------------------------------------------------//
     struct SRawConnectionData
     {
-        SRawConnectionData(XmlNodeRef node, bool isValid)
-            : m_xmlNode(node)
-            , m_isValid(isValid)
-        {}
+        SRawConnectionData(AZ::rapidxml::xml_node<char>* node, bool isValid)
+        {
+            m_xmlNode = DeepCopyNode(node);
+            m_isValid = isValid;
+        }
 
-        XmlNodeRef m_xmlNode;
+        AZ::rapidxml::xml_node<char>* m_xmlNode{ nullptr };
 
         // indicates if the connection is valid for the currently loaded middleware
-        bool m_isValid;
+        bool m_isValid{ false };
+
+        // Rapid XML provides a 'clone_node' utility that will copy an entire node tree,
+        // but it only copies pointers of any strings in the node names and values.
+        // This causes problems with storing raw xml nodes as this class does because strings
+        // will be pointing into the memory pool of an xml document that has gone out of scope.
+        // This function is a rewritten version of 'clone_node' that does the deep copy of strings
+        // into the new destination tree.
+        static AZ::rapidxml::xml_node<char>* DeepCopyNode(AZ::rapidxml::xml_node<char>* srcNode)
+        {
+            AZ::rapidxml::xml_node<char>* destNode = nullptr;
+            if (srcNode)
+            {
+                XmlAllocator& xmlAlloc(AudioControls::s_xmlAllocator);
+                destNode = xmlAlloc.allocate_node(srcNode->type());
+
+                destNode->name(xmlAlloc.allocate_string(srcNode->name(), srcNode->name_size()), srcNode->name_size());
+                destNode->value(xmlAlloc.allocate_string(srcNode->value(), srcNode->value_size()), srcNode->value_size());
+
+                for (AZ::rapidxml::xml_node<char>* child = srcNode->first_node(); child != nullptr; child = child->next_sibling())
+                {
+                    destNode->append_node(DeepCopyNode(child));
+                }
+
+                for (AZ::rapidxml::xml_attribute<char>* attr = srcNode->first_attribute(); attr != nullptr; attr = attr->next_attribute())
+                {
+                    destNode->append_attribute(xmlAlloc.allocate_attribute(
+                        xmlAlloc.allocate_string(attr->name(), attr->name_size()),
+                        xmlAlloc.allocate_string(attr->value(), attr->value_size()),
+                        attr->name_size(),
+                        attr->value_size()
+                    ));
+                }
+            }
+
+            return destNode;
```

Comment:
I think a safer API is to return a unique_ptr of `rapidxml:xml_node`, and then just invoke [unique_ptr::release](https://en.cppreference.com/w/cpp/memory/unique_ptr/release) it when it gets copied.
That way if the `rapidxml::xml_node` isn't stored into a variable it's memory gets cleaned up

Also you may want to mark this function as [[[nodiscard]]](https://en.cppreference.com/w/cpp/language/attributes/nodiscard) as the return value is expected to be handled by the user.
This avoids the user calling the function, but not using the copied node.

Reply:
I'm not sure if using `unique_ptr` would work.  Since this is a recursive function, line 56 would need to be something like `append_node(DeepCopyNode(child).get())`.  I think we would have temporary unique_ptrs being created and destroyed as it's trying to build the destination tree.

I will add `[[nodiscard]]`.

Reply:
I see what you're saying about calling `.release()` to solve that issue.  I will also look at where I store these nodes and convert them to use `unique_ptr` as well.